File: schoolbookRSA.py
#this is schoolbook RSA and should not be used in praxis (e.g. padded must be added...)
#this version ist for demonstration purposes only
plain = 6 #text as number to encrypt

p = 11 #choose to primes, both need to be kept secret
q = 7
n = p*q 
phi = (p-1)*(q-1)
pubkey = 7 #gcd(pubkey,phi)=1
privkey = 43 #pubkey*privkey mod phi = 1

print("plain: " + str(plain))
cipher = plain**pubkey % n
print("cipher: " + str(cipher))
text = cipher**privkey % n
print("plain again:" + str(text))

File: SchoolbookRSA.java
import java.math.BigInteger;


public class SchoolbookRSA 
{
	
	
	public static BigInteger crypt(String message, BigInteger exp, BigInteger modulo) 
	{
		BigInteger m = new BigInteger(message.getBytes());
		return m.modPow(exp, modulo);
	}
	
	public static void main(String[] args) 
	{
		//Schoolbook RSA should not be used because it is not really safe!
		//For realistic texts the modulo needs to be much bigger...
		BigInteger pubKey = new BigInteger("7");
		BigInteger privKey = new BigInteger("43");
		BigInteger p = new BigInteger("7");
		BigInteger q = new BigInteger("11");
		BigInteger modulo = p.multiply(q);
		String plain = "5";
		String cipher = new String(crypt(plain, pubKey, modulo).toByteArray());
		String plainagain = new String(crypt(cipher, privKey, modulo).toByteArray());
		System.out.println("plain: " + plain);
		System.out.println("cipher: " + crypt(plain, pubKey, modulo));
		System.out.println("plain again: " + plainagain);
	}

}
